apps/hannk/interpreter/transforms.cpp

```diff
@@ -25,28 +27,28 @@ T *cast_op(Op *x) {
     }
 }
 
-}  // namespace
-
-void remove_dead_ops(OpGroup *root) {
+void remove_dead_ops_impl(OpGroup *root, const std::set<Tensor *> &root_outputs) {
```

Comment:
Is root_outputs equivalent to root->output(i) for i = 0...op->output_count()? I think I have a helper function somewhere that checks if a tensor is in the inputs or outputs of an op, maybe we should use that instead of passing this set.